From 6eb32636cd2ccfb9bc2bec49deb363db3c78ea6f Mon Sep 17 00:00:00 2001
From: Suleyman TURKMEN <sturkmen@hotmail.com>
Date: Wed, 29 Jan 2025 08:26:48 +0300
Subject: [PATCH] Update OpenCVFindAVIF.cmake

---
 cmake/OpenCVFindAVIF.cmake | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/cmake/OpenCVFindAVIF.cmake b/cmake/OpenCVFindAVIF.cmake
index 26195a776965..29413dcbf043 100644
--- a/cmake/OpenCVFindAVIF.cmake
+++ b/cmake/OpenCVFindAVIF.cmake
@@ -19,9 +19,13 @@ if(TARGET avif)
     MARK_AS_ADVANCED(AVIF_LIBRARY)
 
     SET(AVIF_FOUND TRUE)
-    GET_TARGET_PROPERTY(AVIF_LIBRARY avif LOCATION)
+    SET(AVIF_LIBRARY avif)
     GET_TARGET_PROPERTY(AVIF_INCLUDE_DIR1 avif INCLUDE_DIRECTORIES)
     GET_TARGET_PROPERTY(AVIF_INCLUDE_DIR2 avif INTERFACE_INCLUDE_DIRECTORIES)
+    GET_TARGET_PROPERTY(AVIF_DEPENDENCIES avif INTERFACE_LINK_LIBRARIES)
+    if(AVIF_DEPENDENCIES)
+        LIST(INSERT AVIF_LIBRARY 0 ${AVIF_DEPENDENCIES})
+    endif()
     set(AVIF_INCLUDE_DIR)
     if(AVIF_INCLUDE_DIR1)
         LIST(APPEND AVIF_INCLUDE_DIR ${AVIF_INCLUDE_DIR1})
