diff --git modules/imgproc/src/color.hpp modules/imgproc/src/color.hpp
index 70e784427..69afc083e 100644
--- modules/imgproc/src/color.hpp
+++ modules/imgproc/src/color.hpp
@@ -135,6 +135,7 @@ inline int dstChannels(int code)
 {
     switch( code )
     {
+        case COLOR_BayerBG2BGRA: case COLOR_BayerGB2BGRA: case COLOR_BayerRG2BGRA: case COLOR_BayerGR2BGRA:
         case COLOR_BGR2BGRA: case COLOR_RGB2BGRA: case COLOR_BGRA2RGBA:
         case COLOR_BGR5652BGRA: case COLOR_BGR5552BGRA: case COLOR_BGR5652RGBA: case COLOR_BGR5552RGBA:
         case COLOR_GRAY2BGRA:
@@ -145,6 +146,7 @@ inline int dstChannels(int code)
 
             return 4;
 
+        case COLOR_BayerBG2BGR: case COLOR_BayerGB2BGR: case COLOR_BayerRG2BGR: case COLOR_BayerGR2BGR:
         case COLOR_BGRA2BGR: case COLOR_RGBA2BGR: case COLOR_RGB2BGR:
         case COLOR_BGR5652BGR: case COLOR_BGR5552BGR: case COLOR_BGR5652RGB: case COLOR_BGR5552RGB:
         case COLOR_GRAY2BGR:
