From 799172f727c90c618941abcd30df8a4e705fb7d4 Mon Sep 17 00:00:00 2001
From: Maksim Shabunin <maksim.shabunin@gmail.com>
Date: Wed, 22 May 2019 19:14:29 +0300
Subject: [PATCH 1/1] Added ocv_update_file cmake function

---
 cmake/OpenCVUtils.cmake | 18 ++++++++++++++++++
 1 file changed, 18 insertions(+)

diff --git a/cmake/OpenCVUtils.cmake b/cmake/OpenCVUtils.cmake
index 9cde176..7efef0d 100644
--- a/cmake/OpenCVUtils.cmake
+++ b/cmake/OpenCVUtils.cmake
@@ -1769,3 +1769,21 @@ macro(ocv_git_describe var_name path)
     set(${var_name} "unknown")
   endif()
 endmacro()
+
+# ocv_update_file(filepath content [VERBOSE])
+# - write content to file
+# - will not change modification time in case when file already exists and content has not changed
+function(ocv_update_file filepath content)
+  if(EXISTS "${filepath}")
+    file(READ "${filepath}" actual_content)
+  else()
+    set(actual_content "")
+  endif()
+  if("${actual_content}" STREQUAL "${content}")
+    if(";${ARGN};" MATCHES ";VERBOSE;")
+      message(STATUS "${filepath} contains the same content")
+    endif()
+  else()
+    file(WRITE "${filepath}" "${content}")
+  endif()
+endfunction()
-- 
2.9.2.windows.1

