--- work/modules/dnn/src/caffe/caffe_io.cpp.orig	2018-08-07 10:01:38.000000000 +0100
+++ work/modules/dnn/src/caffe/caffe_io.cpp	2018-08-07 10:02:21.000000000 +0100
@@ -1120,7 +1120,7 @@
     std::ifstream fs(filename, std::ifstream::in);
     CHECK(fs.is_open()) << "Can't open \"" << filename << "\"";
     IstreamInputStream input(&fs);
-    return google::protobuf::TextFormat::Parser(true).Parse(&input, proto);
+    return google::protobuf::TextFormat::Parser().Parse(&input, proto);
 }
 
 bool ReadProtoFromBinaryFile(const char* filename, Message* proto) {
