--- work/cmake/OpenCVModule.cmake.orig	2019-02-15 10:18:07.235333424 +0000
+++ work/cmake/OpenCVModule.cmake	2019-02-15 10:18:54.365082507 +0000
@@ -524,7 +524,7 @@
     endforeach()
   endwhile()
 
-#  message(STATUS "List of active modules: ${OPENCV_MODULES_BUILD}")
+  message(STATUS "List of active modules: ${OPENCV_MODULES_BUILD}")
 
   foreach(m ${OPENCV_MODULES_BUILD})
     set(deps_${m} ${OPENCV_MODULE_${m}_REQ_DEPS})
@@ -535,7 +535,7 @@
         endif()
       endif()
     endforeach()
-#    message(STATUS "Initial deps of ${m} (w/o private deps): ${deps_${m}}")
+    message(STATUS "Initial deps of ${m} (w/o private deps): ${deps_${m}}")
   endforeach()
 
   # propagate dependencies
@@ -547,7 +547,7 @@
         if((NOT m STREQUAL m2) AND ";${deps_${m}};" MATCHES ";${m2};")
           foreach(d ${deps_${m2}})
             if(NOT (";${deps_${m}};" MATCHES ";${d};"))
-#              message(STATUS "  Transfer dependency ${d} from ${m2} to ${m}")
+              message(STATUS "  Transfer dependency ${d} from ${m2} to ${m}")
               list(APPEND deps_${m} ${d})
               set(has_changes ON)
             endif()
@@ -557,7 +557,7 @@
                 AND OPENCV_MODULE_${m2}_IS_PART_OF_WORLD
                 AND NOT OPENCV_MODULE_${m}_IS_PART_OF_WORLD)
               if(NOT (";${deps_${m}};" MATCHES ";opencv_world;"))
-#                message(STATUS "  Transfer dependency opencv_world alias ${m2} to ${m}")
+                message(STATUS "  Transfer dependency opencv_world alias ${m2} to ${m}")
                 list(APPEND deps_${m} opencv_world)
                 set(has_changes ON)
               endif()
@@ -587,7 +587,7 @@
   ocv_list_sort(OPENCV_MODULES_BUILD)
 
   foreach(m ${OPENCV_MODULES_BUILD})
-#    message(STATUS "FULL deps of ${m}: ${deps_${m}}")
+    message(STATUS "FULL deps of ${m}: ${deps_${m}}")
     set(OPENCV_MODULE_${m}_DEPS ${deps_${m}})
     set(OPENCV_MODULE_${m}_DEPS_EXT ${deps_${m}})
     ocv_list_filterout(OPENCV_MODULE_${m}_DEPS_EXT "^opencv_[^ ]+$")
@@ -626,10 +626,10 @@
     set(OPENCV_MODULE_${m}_DEPS_EXT ${OPENCV_MODULE_${m}_DEPS_EXT} CACHE INTERNAL "Extra dependencies of ${m} module")
     set(OPENCV_MODULE_${m}_DEPS_TO_LINK ${LINK_DEPS} CACHE INTERNAL "Flattened dependencies of ${m} module (for linker)")
 
-#    message(STATUS "  module deps of ${m}: ${OPENCV_MODULE_${m}_DEPS}")
-#    message(STATUS "  module link deps of ${m}: ${OPENCV_MODULE_${m}_DEPS_TO_LINK}")
-#    message(STATUS "  extra deps of ${m}: ${OPENCV_MODULE_${m}_DEPS_EXT}")
-#    message(STATUS "")
+    message(STATUS "  module deps of ${m}: ${OPENCV_MODULE_${m}_DEPS}")
+    message(STATUS "  module link deps of ${m}: ${OPENCV_MODULE_${m}_DEPS_TO_LINK}")
+    message(STATUS "  extra deps of ${m}: ${OPENCV_MODULE_${m}_DEPS_EXT}")
+    message(STATUS "")
   endforeach()
 
   __ocv_sort_modules_by_deps(OPENCV_MODULES_BUILD)
