--- modules/imgproc/src/drawing.cpp
+++ modules/imgproc/src/drawing.cpp
@@ -1023,7 +1023,7 @@ EllipseEx( Mat& img, Point2l center, Size2l axes,
            int angle, int arc_start, int arc_end,
            const void* color, int thickness, int line_type )
 {
-    axes.width = std::abs(axes.width), axes.height = std::abs(axes.height);
+    axes.width = cv::abs(axes.width), axes.height = cv::abs(axes.height);
     int delta = (int)((std::max(axes.width,axes.height)+(XY_ONE>>1))>>XY_SHIFT);
     delta = delta < 3 ? 90 : delta < 10 ? 30 : delta < 15 ? 18 : 5;
